test/server_socket_ethernet_test/StreamPacket.h

```diff
@@ -0,0 +1,54 @@
+/*
+Smooth - A C++ framework for embedded programming on top of Espressif's ESP-IDF
+Copyright 2019 Per Malmberg (https://gitbub.com/PerMalmberg)
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#pragma once
+#include "smooth/core/network/IPacketDisassembly.h"
+#include <array>
+
+class StreamPacket
+    : public smooth::core::network::IPacketDisassembly
+{
+    public:
+        StreamPacket() = default;
+
+        StreamPacket(const StreamPacket&) = default;
+
+        StreamPacket& operator=(const StreamPacket&) = default;
+
+        explicit StreamPacket(char c)
+        {
+            buff[0] = static_cast<unsigned char>(c);
+        }
+
+        int get_send_length() override
+        {
+            return static_cast<int>(buff.size());
+        }
+
+        const uint8_t* get_data() override
+        {
+            return buff.data();
+        }
+
+        std::array<uint8_t, 1>& data()
+        {
+            return buff;
+        }
+
+    private:
+        std::array<uint8_t, 1> buff{};
+};
```

Comment:
The ethernet test looks like a copy of the Wifi test. Can we do something better than to duplicate the entire code base for the test so that future improvements don't have to be made for both?

Reply:
I copy-pasted to not affect your test.
Ethernet could be included based on #ifdefs by kconfig in the existing test too.